lib/mayaHydra/hydraExtensions/sceneIndex/mayaHydraSceneIndex.cpp

```diff
@@ -476,16 +476,6 @@ MayaHydraSceneIndex::MayaHydraSceneIndex(
 }
 
 MayaHydraSceneIndex::~MayaHydraSceneIndex()
-{
-    //If you get a crash in a callback with a nullptr for _sceneIndex, 
-    // it may be due to the fact that the _sceneIndex pointer has been nulled as its ref count reached 0 but the destructor is still being called.
-    //You should call RemoveCallbacksAndDeleteAdapters(); before the destructor is called.
-
-    // Remove our pick handler from the pick handler registry.
-    TF_AXIOM(MayaHydra::PickHandlerRegistry::Instance().Unregister(_rprimPath));
-}
-
-void MayaHydraSceneIndex::RemoveCallbacksAndDeleteAdapters()
```

Comment:
Moved the body of `RemoveCallbacksAndDeleteAdapters` into the `MayaHydraSceneIndex` destructor itself.